frog/imports/client/SingleActivity/Finish.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import React from 'react';
+import Card from '@material-ui/core/Card';
+import IconButton from '@material-ui/core/IconButton';
+import ArrowBack from '@material-ui/icons/ArrowBack';
+import Typography from '@material-ui/core/Typography';
+import { type PropsT } from './types';
+
+/**
+ * The final screen displayed after the creation of a single activity.
+ * @param {Object} url - An object containing the paths of public and dashboard view
+ * @param {Function} onReturn - Callback function for handling the 'Back' button
+ */
+export default function Finish(
+  props: {
+    url: {
+      public: String,
+      dashboard: String
+    },
+    onReturn: Function
+  } & PropsT
+) {
+  const { url, classes } = props;
+  return (
+    <Card raised className={classes.card}>
+      <IconButton
+        onClick={() => props.onReturn()}
+        className={classes.back_button}
+      >
+        <ArrowBack />
+      </IconButton>
+      <Typography variant="h3" component="h2" className={classes.padded_text}>
+        You're all set, please share this link with the participants:{' '}
+        <a
```

Comment:
Can you extract all the URL calculation logic into a separate function? In fact it could even be in frog-utils, but at least it shouldn't be inthe JSX and it shouldn't be repeated.

Reply:
I'll currently extract it into a variable in the same file but later on, it could be moved into `frog-utils`